hunit/Test/Tasty/HUnit/Orig.hs

```diff
@@ -133,15 +129,33 @@ instance (AssertionPredicable t) => AssertionPredicable (IO t)
 
 
 -- | Exception thrown by 'assertFailure' etc.
-data HUnitFailure = HUnitFailure (Maybe SrcLoc) String
+data HUnitFailure = HUnitFailure CallStack String
     deriving (Eq, Show, Typeable)
 instance E.Exception HUnitFailure
 
-prependLocation :: Maybe SrcLoc -> String -> String
-prependLocation mbloc s =
-  case mbloc of
-    Nothing -> s
-    Just loc -> srcLocFile loc ++ ":" ++ show (srcLocStartLine loc) ++ ":\n" ++ s
+prependCallStack :: CallStack -> String -> String
+prependCallStack cs s =
+  "Error message: " <> s <> "\n\n" <> prettyCallStack cs
+
+prettyCallStack :: CallStack -> String
+prettyCallStack = intercalate "\n" . prettyCallStackLines
+
+prettyCallStackLines :: CallStack -> [String]
+prettyCallStackLines cs = case cs of
+  []  -> []
+  stk -> "CallStack (from HasCallStack):"
+       : map (("  " ++) . prettyCallSite) stk
+  where
+    prettyCallSite (f, loc) = f ++ ", called at " ++ prettySrcLoc loc
```

Comment:
`base` re-exports `prettyCallStackLines`, so we should be able to use that. I'm assuming you made a local definition because you didn't want to rely on the `ghc` package?

Reply:
this actually has a different reason, namely that the `CallStack` here is not the same that is used in `GHC.Exception`.